packages/gotrue/lib/src/fetch.dart

```diff
@@ -28,9 +31,12 @@ class GotrueFetch {
     return error.toString();
   }
 
-  String? _getErrorCode(dynamic error) {
+  String? _getErrorCode(dynamic error, String key) {
     if (error is Map) {
-      return error['error_code']?.toString();
+      final dynamic errorCode = error[key];
+      if (errorCode is String) {
+        return errorCode;
+      }
```

Comment:
Would there be any cases where the error code is a number? HTTP Error 400, for example.

Reply:
It should not.

For the old response, the `error_code` is declared as a `string`:
- https://github.com/supabase/auth/blob/701a779cf092e777dd4ad4954dc650164b09ab32/internal/api/errors.go#L103

For the 2024-01-01 version of the API, it will be an `ErrorCode`, which is also a `string`:
- https://github.com/supabase/auth/blob/701a779cf092e777dd4ad4954dc650164b09ab32/internal/api/errors.go#L182

Anyway, I could be missing something :)